### PR TITLE
Convert libvirt template to workflow

### PR DESCRIPTION
**Pull request by sallyom**
Picking up @praveenkumar 's PR to rebase/test/add ssh config vars

### CODE REVIEW

File: ci-operator/step-registry/openshift/e2e/libvirt/test/openshift-e2e-libvirt-test-commands.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -euo pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+INSTANCE_PREFIX="${NAMESPACE}"-"${JOB_NAME_HASH}"
+
+mkdir -p "${HOME}"/.ssh
```
Comment (dobbymoodge) [inline ci-operator/step-registry/openshift/e2e/libvirt/test/openshift-e2e-libvirt-test-commands.sh:8]: Remember to `chmod 0700 "${HOME}/.ssh`

Comment (sallyom) [reply ci-operator/step-registry/openshift/e2e/libvirt/test/openshift-e2e-libvirt-test-commands.sh]: ok, thank you, will do

File: ci-operator/step-registry/openshift/e2e/gcp/libvirt/test/openshift-e2e-gcp-libvirt-test-commands.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+# TODO: ADD TESTS - HOW TO ADD openshift-tests binary? 
+cat  > "${HOME}"/run-tests.sh << 'EOF'
+#!/bin/bash
+set -euo pipefail
+export PATH=/home/packer:$PATH
+mkdir -p /tmp/artifacts/junit
+function run-tests() {
+  # Check the cluster operator output, status for available should be true for all operators
+  oc get co -ojsonpath='{.items[*].status.conditions[?(@.type=="Available")].status}' | grep -v True
```
Comment (tnozicka) [inline ci-operator/step-registry/openshift/e2e/gcp/libvirt/test/openshift-e2e-gcp-libvirt-test-commands.sh:36]: none should be progressing or degraded too, but I haven't checked how uch is this verified in regular e2e or it it just relies on the installer

Comment (sallyom) [reply ci-operator/step-registry/openshift/e2e/gcp/libvirt/test/openshift-e2e-gcp-libvirt-test-commands.sh:36]: for now, I'm simply trying to get the test pod to run :)  I have to fix this up

Comment (sallyom) [reply ci-operator/step-registry/openshift/e2e/gcp/libvirt/test/openshift-e2e-gcp-libvirt-test-commands.sh]: updated, added the same tests as are currently running w/ the template

**Comment by sallyom** [comment]
@praveenkumar it's ready!! I added the same set of tests that are currently running in the openshift/installer libvirt-e2e job.

**Comment by sallyom** [comment]
@praveen, after this merges, we can look at replacing the custom image with a provision script within the `upi-gcp-nested-pre` workflow.

File: ci-operator/step-registry/openshift/e2e/gcp/libvirt/conf/OWNERS [hunk 3]
```diff
@@ -0,0 +1,7 @@
+approvers:
```
Comment (wking) [inline ci-operator/step-registry/openshift/e2e/gcp/libvirt/conf/OWNERS:1]: nit: might be easier to maintain if you use a symlink here; compare #12832.

File: ci-operator/step-registry/openshift/e2e/gcp/libvirt/conf/openshift-e2e-gcp-libvirt-conf-commands.sh [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] subPath should be able to unmount after the subpath directory is deleted [Suite:openshift/conformance/parallel] [Suite:k8s]
+[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] subPath should support existing single file [Suite:openshift/conformance/parallel] [Suite:k8s]
+[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] subPath should support file as subpath [Suite:openshift/conformance/parallel] [Suite:k8s]
+[sig-storage] In-tree Volumes [Driver: hostPath] [Testpattern: Inline-volume (default fs)] subPath should support non-existent path [Suite:openshift/conformance/parallel] [Suite:k8s]
+[sig-storage] Secrets should be consumable from pods in volume with mappings [NodeConformance] [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
+[sig-storage] Subpath Atomic writer volumes should support subpaths with downward pod [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
+#
+EOF
+
+cat <(echo "$INCL") > "${SHARED_DIR}/test-list"
```
Comment (wking) [inline ci-operator/step-registry/openshift/e2e/gcp/libvirt/conf/openshift-e2e-gcp-libvirt-conf-commands.sh:144]: nit: no need to go through `INCL`, is there?  You can just:

```sh
cat <<EOF > "${SHARED_DIR}/test-list"
[Conformance][Area:Networking][Feature:Router] The HAProxy router should override the route host for overridden domains with a custom value [Suite:openshift/conformance/parallel/minimal]
...
EOF

Comment (sallyom) [reply ci-operator/step-registry/openshift/e2e/gcp/libvirt/conf/openshift-e2e-gcp-libvirt-conf-commands.sh]: right, updated, thanks

File: ci-operator/step-registry/openshift/e2e/gcp/libvirt/conf/OWNERS (hunk 3 above)
Comment (sallyom) [reply ci-operator/step-registry/openshift/e2e/gcp/libvirt/conf/OWNERS:1]: deduped ci-op/step-registry/openshfit/e2e/gcp/libvirt, thanks! did not know of this script b4 : )

**Comment by sallyom** [comment]
thanks for the review @wking, updated, ptal!

**Pull request merged**